plugins/eni/engine/dhclient.go

```diff
@@ -131,14 +144,29 @@ func (client *dhclient) IsExecutableInPath() bool {
 
 func (client *dhclient) Start(deviceName string, ipRev int) error {
 	args := client.constructDHClientArgs(deviceName, ipRev)
-	cmd := client.exec.Command(client.executable, args...)
-	out, err := cmd.CombinedOutput()
-	if err != nil {
-		log.Errorf("Error executing '%s' with args '%v': raw output: %s",
-			client.executable, args, string(out))
-		return errors.Wrapf(err,
-			"engine dhclient: unable to start dhclient for ipv%d address; command: %s %v; output: %s",
-			ipRev, client.executable, args, string(out))
+	var out []byte
+	var err error
+	attempts := 1
+	for {
+		cmd := client.exec.Command(client.executable, args...)
+		out, err = cmd.CombinedOutput()
+		if err == nil {
+			return nil
+		}
+
+		log.Errorf("Error executing '%s' with args '%v' (attempt %d/%d): raw output: %s",
```

Comment:
Should this be an ERROR or a WARN?

Reply:
`Warn` would be better. I'll modify that.